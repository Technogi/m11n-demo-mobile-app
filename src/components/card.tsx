import React from 'react'
import { View, StyleSheet, ViewStyle } from 'react-native'
import { moderateScale } from 'react-native-size-matters'

/**
 * Returns a card component.
 *
 * @param {({
 *   children?: JSX.Element | JSX.Element[]
 *   style?: ViewStyle
 * })} {
 *   children: child component,
 *   style: view style
 * }
 * @returns {JSX.Element} JSX.Element
 */
const Card = ({ children, style }: { children?: JSX.Element | JSX.Element[]; style?: ViewStyle }): JSX.Element => {
  const { viewStyle } = styles

  return <View style={{ ...viewStyle, ...style }}>{children}</View>
}

const styles = StyleSheet.create({
  viewStyle: {
    backgroundColor: 'white',
    width: '100%',
    minHeight: 60,
    borderRadius: moderateScale(10),
    padding: moderateScale(15),
  },
})

export default Card
