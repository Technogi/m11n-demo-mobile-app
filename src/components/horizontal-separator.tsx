import React from 'react'
import { View } from 'react-native'

/**
 * Returns a horizontal separator that can be used to give space between components.
 *
 * @param {{ space: number }} { space }
 * @return {JSX.Element}  {JSX.Element}
 */
const HorizontalSeparator = ({ space }: { space: number }): JSX.Element => (
  <View style={{ borderBottomWidth: 0.5, borderBottomColor: '#939393', marginTop: space }} />
)

export default HorizontalSeparator
