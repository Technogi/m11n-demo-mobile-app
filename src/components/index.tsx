import Container from './layout/container'
import Content from './layout/content'
import Card from './card'
import PressableOpacity from './buttons/pressable-opacity'
import Button from './buttons/button'
import toastConfig from './toast/toast-config'
import Headline from './text/headline'
import Subtitle from './text/subtitle'
import Body from './text/body'
import Caption from './text/caption'
import CustomTextInput from './forms/text-input/custom-text-input'
import TextInputForm from './forms/text-input/text-input.form'
import SecondaryButton from './buttons/secondary.button'
import LogoHeader from './logo-header'
import TitleHeader from './title-header'

import { ToastType } from './toast/types'

export {
  Container,
  Content,
  Card,
  PressableOpacity,
  Button,
  SecondaryButton,
  toastConfig,
  ToastType,
  Headline,
  Subtitle,
  Body,
  Caption,
  CustomTextInput,
  TextInputForm,
  LogoHeader,
  TitleHeader,
}
