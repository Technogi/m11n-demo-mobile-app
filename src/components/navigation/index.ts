import MainTabIcon from './main-tab-icon'

export { MainTabIcon }
