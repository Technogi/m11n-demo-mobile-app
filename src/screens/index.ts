import HomeScreen from './home/home.screen'
import LoginScreen from './login/login.screen'
import SalesScreen from './sales/sales.screen'
import SalesForecastsScreen from './sales-forecasts/sales-forecasts.screen'

export default {
  LoginScreen,
  HomeScreen,
  SalesScreen,
  SalesForecastsScreen,
}
