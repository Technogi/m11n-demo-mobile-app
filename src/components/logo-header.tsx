import { StyleSheet, View, ViewStyle } from 'react-native'
import React from 'react'
import { moderateScale } from 'react-native-size-matters'
import MaterialIcons from 'react-native-vector-icons/MaterialIcons'

import { theme } from 'src/styles'

import Headline from './text/headline'

const LogoHeader = ({ containerStyle }: { containerStyle?: ViewStyle }): JSX.Element => {
  const { titleStyle, headlineStyle } = styles

  return (
    <View style={{ ...titleStyle, ...containerStyle }}>
      <MaterialIcons name="analytics" color={theme.PRIMARY_COLOR} size={theme.ICON_SIZE_7XL} />
      <Headline style={headlineStyle}>POStats</Headline>
    </View>
  )
}

const styles = StyleSheet.create({
  titleStyle: {
    marginVertical: moderateScale(20),
    flexDirection: 'row',
    alignItems: 'center',
  },
  headlineStyle: {
    fontSize: theme.FONT_SIZE_3XL,
    marginLeft: moderateScale(10),
  },
})

export default LogoHeader
