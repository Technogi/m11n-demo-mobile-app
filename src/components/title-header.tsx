import { StyleSheet, View, ViewStyle } from 'react-native'
import React from 'react'
import { moderateScale } from 'react-native-size-matters'
import Feather from 'react-native-vector-icons/Feather'
import { NativeStackNavigationProp } from '@react-navigation/native-stack'

import { theme } from 'src/styles'

import Headline from './text/headline'
import PressableOpacity from './buttons/pressable-opacity'

const TitleHeader = ({
  title,
  containerStyle,
  navigation,
}: {
  title: string
  containerStyle?: ViewStyle
  navigation: NativeStackNavigationProp<any, any>
}): JSX.Element => {
  const { titleStyle, headlineStyle } = styles

  return (
    <View style={{ ...titleStyle, ...containerStyle }}>
      <PressableOpacity hitSlop={5} containerStyle={{ flex: 0.1 }} onPress={() => navigation.goBack()}>
        <Feather name="chevron-left" color={theme.PRIMARY_COLOR} size={theme.ICON_SIZE_2XL} />
      </PressableOpacity>

      <View style={{ flex: 0.9 }}>
        <Headline style={headlineStyle}>{title}</Headline>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  titleStyle: {
    marginVertical: moderateScale(20),
    flexDirection: 'row',
    alignItems: 'center',
  },
  headlineStyle: {
    marginLeft: moderateScale(10),
  },
})

export default TitleHeader
