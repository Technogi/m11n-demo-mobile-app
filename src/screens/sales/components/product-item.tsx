import { StyleSheet, View } from 'react-native'
import React, { useContext } from 'react'
import { moderateScale } from 'react-native-size-matters'
import CircularProgress from 'react-native-circular-progress-indicator'

import { formatNumber } from 'src/utils/helpers'
import { mainStyle, theme } from 'src/styles'
import { LocalizationContext } from 'src/context/localization/localization.context'

import { Body, Card, PressableOpacity } from 'src/components'
import { Product } from '../models/product.model'

const ProductItem = ({ item }: { item: Product }): JSX.Element => {
  const { name, price, sales: productSales, total, performance } = item

  const {
    translations: { sales },
  } = useContext(LocalizationContext)

  const { cardContainer } = styles

  const performanceFormatted = Math.floor(performance * 100)

  const getCircleColor = (value: number): string => {
    let color = '#fcc735'

    if (value < 0.3) color = '#fc6335'
    if (value > 0.7) color = '#34eb83'

    return color
  }

  return (
    <Card style={cardContainer}>
      <PressableOpacity containerStyle={{ flexDirection: 'row' }} onPress={() => {}}>
        <View style={{ flex: 0.3 }}>
          <CircularProgress
            value={performanceFormatted}
            radius={moderateScale(35)}
            progressValueColor={theme.BODY_TEXT_COLOR}
            progressValueStyle={mainStyle.textLgStyle()}
            activeStrokeColor={getCircleColor(performance)}
            inActiveStrokeColor={getCircleColor(performance)}
            inActiveStrokeOpacity={0.2}
          />
        </View>

        <View style={{ flex: 0.7 }}>
          <Body bold style={{ color: theme.PRIMARY_COLOR }}>
            {name}
          </Body>

          <Body style={{ marginTop: moderateScale(5) }}>
            <Body bold>{sales.price}: </Body>${formatNumber(Number(price.toFixed(2)))}
          </Body>

          <Body>
            <Body bold>{sales.noSales}: </Body>
            {formatNumber(productSales)}
          </Body>

          <Body>
            <Body bold>{sales.totalSold}: </Body>${formatNumber(Number(total.toFixed(2)))}
          </Body>
        </View>
      </PressableOpacity>
    </Card>
  )
}

const styles = StyleSheet.create({
  cardContainer: {
    justifyContent: 'center',
    padding: moderateScale(15),
    marginBottom: moderateScale(15),
  },
})

export default ProductItem
