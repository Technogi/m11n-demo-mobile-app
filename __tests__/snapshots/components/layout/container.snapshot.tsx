import React from 'react'
import { View } from 'react-native'
import renderer from 'react-test-renderer'
import { Container } from 'src/components'

it('Container renders correctly', () => {
  const tree = renderer
    .create(
      <Container>
        <View />
      </Container>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
