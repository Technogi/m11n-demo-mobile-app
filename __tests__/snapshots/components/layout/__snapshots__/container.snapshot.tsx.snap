// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F3F5F8",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
